src/main/java/org/wso2/carbon/connector/iso8583/ISO8583MessageHandler.java

```diff
@@ -56,28 +57,31 @@ public ISO8583MessageHandler(MessageContext messageContext, String details, Stri
     /**
      * handle the iso8583 message request and responses
      *
-     * @param isoMessage  packed ISOMessage
-     * @param connection  Socket connection with backend Test server
      * @param messageContext the message context
+     * @param connection  Socket connection with backend Test server
+     * @param isoMessage  packed ISOMessage
      */
-    public void clientHandler(MessageContext messageContext, Socket connection, String isoMessage) {
+    public void clientHandler(MessageContext messageContext, Socket connection, byte[] isoMessage) {
         DataOutputStream outStream = null;
-        BufferedReader inFromServer = null;
-        String message;
+        DataInputStream inFromServer = null;
         try {
             outStream = new DataOutputStream(connection.getOutputStream());
-            inFromServer = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            inFromServer = new DataInputStream(connection.getInputStream());
```

Comment:
Furthermore it is wrong to close connection in this finally block. It should be closed by the method who creates it in it's finally block.

Reply:
resolved